compiler/vm.nim

```diff
@@ -742,8 +818,18 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
       of rkNode:
         if regs[ra].node.kind == nkNilLit:
           stackTrace(c, tos, pc, errNilAccess)
-        regs[ra].node[] = regs[rc].regToNode[]
-        regs[ra].node.flags.incl nfIsRef
+        let node = regs[ra].node
+        let typ = node.typ
```

Comment:
Same.

Reply:
ditto, added a typ != nil in branch ; IIUC the tests i wrote would've caught it if it were nil